### PR TITLE
Fixed deployment of Elasticsearch via its operator

### LINKED ISSUE #233: Update CR PR broke elasticsearch deployment
https://github.com/jaegertracing/jaeger-operator/pull/193/files#r260355219. Now the code does not create ES CR.

cc @jpkrohling

### PR DESCRIPTION
**Pull request by jpkrohling**
Fixes #233 by adding the ES type to the controller's reconcile loop and inventory.

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

**Comment by jpkrohling** [comment]
This change is [Reviewable](https://reviewable.io/reviews/jaegertracing/jaeger-operator/234)

File: pkg/controller/jaeger/elasticsearch.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	inv := inventory.ForElasticsearches(list.Items, desired)
+	for _, d := range inv.Create {
+		logFields.WithField("elasticsearch", d.Name).Debug("creating config maps")
+		if err := r.client.Create(context.Background(), &d); err != nil {
+			return err
+		}
+	}
+
+	for _, d := range inv.Update {
+		logFields.WithField("configMap", d.Name).Debug("updating config maps")
```
Comment (pavolloffay) [inline pkg/controller/jaeger/elasticsearch.go]: config maps?

File: pkg/controller/jaeger/elasticsearch.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		if err := r.client.Update(context.Background(), &d); err != nil {
+			return err
+		}
+	}
+
+	for _, d := range inv.Delete {
+		logFields.WithField("configMap", d.Name).Debug("deleting config maps")
```
Comment (pavolloffay) [inline pkg/controller/jaeger/elasticsearch.go]: config maps?

**Comment by jpkrohling** [comment]
Running locally with this PR causes this when deploying `deploy/examples/simple-prod-deploy-es.yaml`:

> ERROR: logging before flag.Parse: E0226 18:01:00.096980   28875 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:196: Failed to list *v1alpha1.Elasticsearch: v1.ListOptions is not suitable for converting to "logging.openshift.io/v1alpha1" in scheme "k8s.io/client-go/kubernetes/scheme/register.go:60"

**Comment by pavolloffay** [comment]
To test it make sure that `deploy/examples/simple-prod-deploy-es.yaml` works it should deploy one ES and jaeger components should be up and ready. You can also change the index cleaner schedule time to test it works.

**Comment by jpkrohling** [comment]
I got it working now:

```
$ kubectl get pods
NAME                                                 READY     STATUS      RESTARTS   AGE
elasticsearch-clientdatamaster-0-1-6c7975dd5-n7g49   1/1       Running     0          1h
simple-prod-collector-76c76f74fb-rqvrs               1/1       Running     0          3m
simple-prod-es-index-cleaner-1551271500-7ftk6        0/1       Completed   0          10s
simple-prod-query-7df889c65b-42s47                   2/2       Running     0          3m
$ kubectl logs simple-prod-collector-76c76f74fb-rqvrs
{"level":"info","ts":1551271304.7249413,"caller":"healthcheck/handler.go:99","msg":"Health Check server started","http-port":14269,"status":"unavailable"}
{"level":"info","ts":1551271307.447118,"caller":"collector/main.go:139","msg":"Starting jaeger-collector TChannel server","port":14267}
{"level":"info","ts":1551271307.4489896,"caller":"grpcserver/grpc_server.go:64","msg":"Starting jaeger-collector gRPC server","grpc-port":"14250"}
{"level":"info","ts":1551271307.4490638,"caller":"collector/main.go:153","msg":"Registering metrics handler with HTTP server","route":"/metrics"}
{"level":"info","ts":1551271307.449096,"caller":"collector/main.go:162","msg":"Starting jaeger-collector HTTP server","http-port":14268}
{"level":"info","ts":1551271307.4491127,"caller":"healthcheck/handler.go:133","msg":"Health Check state change","status":"ready"}
{"level":"info","ts":1551271307.475963,"caller":"collector/main.go:237","msg":"Listening for Zipkin HTTP traffic","zipkin.http-port":9411}
```

**Comment by pavolloffay** [comment]
> The collector and query were in a failed state for quite some time, because they don't reconnect to Elasticsearch upon failure, so, I had to kill the pods manually.

K8s should reschedule pods once they fail. At least I was this behavior before update PR merge. I didn't had to kill pods manually - they restarted 2-3 times until ES was in the ready state.

There is an issue https://github.com/jaegertracing/jaeger-operator/issues/216 which handles the initialization properly.

**Comment by jpkrohling** [comment]
> At least I was this behavior before update PR merge

That PR shouldn't change this behavior. The pod was in a "healthy" state, which is why Kubernetes didn't kill it. This is how the logs look like:

```
$ oc get pods
NAME                                                 READY     STATUS    RESTARTS   AGE
...
simple-prod-collector-76c76f74fb-wv8n8               1/1       Running   0          1h
...


$ oc logs simple-prod-collector-76c76f74fb-wv8n8
{"level":"info","ts":1551266878.3367102,"caller":"healthcheck/handler.go:99","msg":"Health Check server started","http-port":14269,"status":"unavailable"}
{"level":"info","ts":1551266882.0825386,"caller":"collector/main.go:139","msg":"Starting jaeger-collector TChannel server","port":14267}
{"level":"info","ts":1551266882.0826762,"caller":"grpcserver/grpc_server.go:64","msg":"Starting jaeger-collector gRPC server","grpc-port":"14250"}
{"level":"info","ts":1551266882.082747,"caller":"collector/main.go:153","msg":"Registering metrics handler with HTTP server","route":"/metrics"}
{"level":"info","ts":1551266882.0827823,"caller":"collector/main.go:162","msg":"Starting jaeger-collector HTTP server","http-port":14268}
{"level":"info","ts":1551266882.0827997,"caller":"healthcheck/handler.go:133","msg":"Health Check state change","status":"ready"}
{"level":"info","ts":1551266882.1220624,"caller":"collector/main.go:237","msg":"Listening for Zipkin HTTP traffic","zipkin.http-port":9411}
{"level":"error","ts":1551271199.0897613,"caller":"spanstore/writer.go:198","msg":"Failed to create index","trace_id":"265b719bdfc33c79","span_id":"5d238c8980fd5616","error":"no available connection: no Elasticsearch node available","errorVerbose":"no Elasticsearch node available\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.init\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:88\ngithub.com/jaegertracing/jaeger/pkg/es.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/plugin/storage/es.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/plugin/storage.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/cmd/env.init\n\t<autogenerated>:1\nmain.init\n\t<autogenerated>:1\nruntime.main\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/proc.go:189\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333\nno available connection\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*Client).next\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:1157\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*Client).PerformRequestWithOptions\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:1254\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*Client).PerformRequest\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:1190\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*IndicesCreateService).Do\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/indices_create.go:112\ngithub.com/jaegertracing/jaeger/pkg/es/wrapper.IndicesCreateServiceWrapper.Do\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/es/wrapper/wrapper.go:105\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).createIndex\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:159\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).WriteSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:135\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:101\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan-fm\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:88\ngithub.com/jaegertracing/jaeger/cmd/collector/app.ChainedProcessSpan.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/model_consumer.go:34\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).processItemFromQueue\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:127\ngithub.com/jaegertracing/jaeger/cmd/collector/app.NewSpanProcessor.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:56\ngithub.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumers.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/queue/bounded_queue.go:65\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333","stacktrace":"github.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).logError\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:198\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).createIndex\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:168\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).WriteSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:135\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:101\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan-fm\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:88\ngithub.com/jaegertracing/jaeger/cmd/collector/app.ChainedProcessSpan.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/model_consumer.go:34\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).processItemFromQueue\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:127\ngithub.com/jaegertracing/jaeger/cmd/collector/app.NewSpanProcessor.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:56\ngithub.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumers.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/queue/bounded_queue.go:65"}
{"level":"error","ts":1551271199.0905058,"caller":"app/span_processor.go:102","msg":"Failed to save span","error":"Failed to create index: no available connection: no Elasticsearch node available","errorVerbose":"no Elasticsearch node available\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.init\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:88\ngithub.com/jaegertracing/jaeger/pkg/es.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/plugin/storage/es.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/plugin/storage.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/cmd/env.init\n\t<autogenerated>:1\nmain.init\n\t<autogenerated>:1\nruntime.main\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/proc.go:189\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333\nno available connection\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*Client).next\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:1157\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*Client).PerformRequestWithOptions\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:1254\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*Client).PerformRequest\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:1190\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*IndicesCreateService).Do\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/indices_create.go:112\ngithub.com/jaegertracing/jaeger/pkg/es/wrapper.IndicesCreateServiceWrapper.Do\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/es/wrapper/wrapper.go:105\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).createIndex\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:159\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).WriteSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:135\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:101\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan-fm\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:88\ngithub.com/jaegertracing/jaeger/cmd/collector/app.ChainedProcessSpan.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/model_consumer.go:34\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).processItemFromQueue\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:127\ngithub.com/jaegertracing/jaeger/cmd/collector/app.NewSpanProcessor.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:56\ngithub.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumers.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/queue/bounded_queue.go:65\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333\nFailed to create index\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).logError\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:199\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).createIndex\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:168\ngithub.com/jaegertracing/jaeger/plugin/storage/es/spanstore.(*SpanWriter).WriteSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/spanstore/writer.go:135\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:101\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan-fm\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:88\ngithub.com/jaegertracing/jaeger/cmd/collector/app.ChainedProcessSpan.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/model_consumer.go:34\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).processItemFromQueue\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:127\ngithub.com/jaegertracing/jaeger/cmd/collector/app.NewSpanProcessor.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:56\ngithub.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumers.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/queue/bounded_queue.go:65\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333","stacktrace":"github.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:102\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).saveSpan-fm\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:88\ngithub.com/jaegertracing/jaeger/cmd/collector/app.ChainedProcessSpan.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/model_consumer.go:34\ngithub.com/jaegertracing/jaeger/cmd/collector/app.(*spanProcessor).processItemFromQueue\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:127\ngithub.com/jaegertracing/jaeger/cmd/collector/app.NewSpanProcessor.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/app/span_processor.go:56\ngithub.com/jaegertracing/jaeger/pkg/queue.(*BoundedQueue).StartConsumers.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/queue/bounded_queue.go:65"}
```

**Comment by pavolloffay** [comment]
The query and collector exit with 1 if there are no ES nodes available. Maybe you should wait longer?

I have tested on minishift and it worked like before. However, I get the following errors when I edit jaeger e.g. (oc edit jaeger simple-prod)

```
make run WATCH_NAMESPACE=myproject                                                                                                                       2:23 
INFO[0000] Versions                                      arch=amd64 operator-sdk=v0.4.1 os=linux version=go1.11.1
INFO[0000] Auto-detected the platform                    platform=openshift
INFO[0000] Starting the Cmd.                            
INFO[0010] Configured Jaeger instance                    instance=simple-prod namespace=myproject
ERRO[0159] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0160] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0161] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0162] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0164] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0165] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0166] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0167] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0168] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0169] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0172] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0177] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject
ERRO[0187] failed to apply the changes                   error="elasticsearches.logging.openshift.io \"elasticsearch\" already exists" instance=simple-prod namespace=myproject

```

**Comment by jpkrohling** [comment]
> The query and collector exit with 1 if there are no ES nodes available. Maybe you should wait longer?

Not sure: it does get into a failed state when the initial connection cannot be made, with logs like this:

```
$ kubectl logs simple-prod-collector-76c76f74fb-55mc2
{"level":"info","ts":1551274615.2542229,"caller":"healthcheck/handler.go:99","msg":"Health Check server started","http-port":14269,"status":"unavailable"}
{"level":"fatal","ts":1551274620.2866693,"caller":"collector/main.go:103","msg":"Failed to init storage factory","error":"failed to create primary Elasticsearch client: health check timeout: Head https://elasticsearch:9200: dial tcp: lookup elasticsearch on 172.30.0.2:53: server misbehaving: no Elasticsearch node available","errorVerbose":"no Elasticsearch node available\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.init\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:88\ngithub.com/jaegertracing/jaeger/pkg/es.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/plugin/storage/es.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/plugin/storage.init\n\t<autogenerated>:1\ngithub.com/jaegertracing/jaeger/cmd/env.init\n\t<autogenerated>:1\nmain.init\n\t<autogenerated>:1\nruntime.main\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/proc.go:189\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333\nhealth check timeout: Head https://elasticsearch:9200: dial tcp: lookup elasticsearch on 172.30.0.2:53: server misbehaving\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.(*Client).startupHealthcheck\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:1116\ngithub.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic%2ev5.NewClient\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/gopkg.in/olivere/elastic.v5/client.go:244\ngithub.com/jaegertracing/jaeger/pkg/es/config.(*Configuration).NewClient\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/pkg/es/config/config.go:97\ngithub.com/jaegertracing/jaeger/plugin/storage/es.(*Factory).Initialize\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/factory.go:80\ngithub.com/jaegertracing/jaeger/plugin/storage.(*Factory).Initialize\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/factory.go:90\nmain.main.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/main.go:102\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).execute\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:762\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:852\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).Execute\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:800\nmain.main\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/main.go:203\nruntime.main\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/proc.go:201\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333\nfailed to create primary Elasticsearch client\ngithub.com/jaegertracing/jaeger/plugin/storage/es.(*Factory).Initialize\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/es/factory.go:82\ngithub.com/jaegertracing/jaeger/plugin/storage.(*Factory).Initialize\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/plugin/storage/factory.go:90\nmain.main.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/main.go:102\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).execute\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:762\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:852\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).Execute\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:800\nmain.main\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/main.go:203\nruntime.main\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/proc.go:201\nruntime.goexit\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/asm_amd64.s:1333","stacktrace":"main.main.func1\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/main.go:103\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).execute\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:762\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).ExecuteC\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:852\ngithub.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra.(*Command).Execute\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/vendor/github.com/spf13/cobra/command.go:800\nmain.main\n\t/home/travis/gopath/src/github.com/jaegertracing/jaeger/cmd/collector/main.go:203\nruntime.main\n\t/home/travis/.gimme/versions/go1.11.1.linux.amd64/src/runtime/proc.go:201"}
```

The case I got into was later on (`span_writer.go` instead of `main.go`). So, ES was probably receiving connections, but not ready yet? Note that the log error was "Failed to create index" and not "no Elasticsearch node available".

**Comment by jpkrohling** [comment]
By the way, the PR has been updated, to fix the error you reported before when doing `kubectl edit jaeger simple-prod`. With that, the update procedure continues. Looks like #235 was a side-effect, as it seems it works:

```
$ kubectl get cronjobs
NAME                             SCHEDULE      SUSPEND   ACTIVE    LAST SCHEDULE   AGE
simple-prod-es-index-cleaner     */5 * * * *   False     1         2m              4m
simple-prod-spark-dependencies   55 23 * * *   False     0         <none>          4m
```

**Comment by pavolloffay** [comment]
thanks @jpkrohling for looking into this! The #235 seems to be a side effect

**Pull request merged**